components/admin/ApplicationsTable.tsx

```diff
@@ -23,29 +23,49 @@ export type Student = {
 
 interface TableProps {
   activeRole?: ApplicantRole;
-  setNumEntries: (tab: number) => void;
-  numEntries?: number;
+  whichChoiceTab?: number;
+  setNumFirstChoiceEntries: (tab: number) => void;
+  numFirstChoiceEntries?: number;
+  setNumSecondChoiceEntries: (tab: number) => void;
+  numSecondChoiceEntries?: number;
 }
 
 const ApplicationsTable: React.FC<TableProps> = ({
   activeRole,
-  setNumEntries,
+  whichChoiceTab,
+  setNumFirstChoiceEntries,
+  setNumSecondChoiceEntries
 }) => {
-  const [applications, setApplications] = useState<any[]>([]);
-
+  const [firstChoiceApplications, setFirstChoiceApplications] = useState<any[]>([]);
+  const [secondChoiceApplications, setSecondChoiceApplications] = useState<any[]>([]);
   useEffect(() => {
     fetchApplicationsByRole();
-  }, [activeRole]);
+  }, [activeRole, whichChoiceTab]);
 
   const fetchApplicationsByRole = async () => {
-    const result = await fetchGraphql(
-      applicationTableQueries.applicationsByRole,
-      {
-        role: activeRole || ApplicantRole.vpe,
-      },
-    );
-    setApplications(result.data.applicationTable);
-    setNumEntries(result.data.applicationTable.length);
+    const currentRole = activeRole || ApplicantRole.vpe;
+    try {
+      const firstChoiceResult = await fetchGraphql(
```

Comment:
see if there is a way to make both of these queries call being called together. Not sure what its called in JS, but in Elixir, there is Task.async() that spawns multiple processes at once.